src/Compilers/VisualBasic/Portable/Emit/SynthesizedPrivateImplementationDetailsSharedConstructor.vb

```diff
@@ -84,14 +84,14 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
             Next
 
             ' Initialize the module version ID (MVID) field. Dynamic instrumentation requires the MVID of the executing module, and this field makes that accessible.
-            ' MVID = Guid.Parse(ModuleVersionIdString)
+            ' MVID = New Guid(ModuleVersionIdString)
 
-            Dim guidParse As MethodSymbol = factory.WellKnownMember(Of MethodSymbol)(WellKnownMember.System_Guid__Parse)
-            If guidParse IsNot Nothing Then
+            Dim guidConstructor As MethodSymbol = factory.WellKnownMember(Of MethodSymbol)(WellKnownMember.System_Guid__ctor)
+            If guidConstructor IsNot Nothing Then
                 body.Add(
                     factory.Assignment(
                        factory.ModuleVersionId(isLValue:=True),
-                       factory.Call(Nothing, guidParse, ImmutableArray.Create(factory.ModuleVersionIdString()))))
+                       factory.[New](guidConstructor, factory.ModuleVersionIdString())))
```

Comment:
This change (on the VB side) seems to affect no tests. Can you make sure a VB test checks the IL for generated instrumentation?